### PR TITLE
Consolidate UI element for device validaiton result

### PR DESCRIPTION
**Pull request by ttanatb**
This PR just consolidates the UI elements between the two different dialogs. Next one will actually add more details to the DeviceValidationResult.

### CODE REVIEW

**Comment by pau-baiget** [comment]
For UI changes it would be great to see snapshots of before/after. Thanks!

**Comment by ttanatb** [comment]
There's no user-facing UI changes, it just consolidates the repeated code in two places where validation status is surfaced (device table and new trace dialog)

File: gapic/src/main/com/google/gapid/models/Devices.java [hunk 1]
```diff
@@ -194,16 +194,11 @@ public void selectReplayDevice(Device.Instance dev) {
...
-    if (fromCache.passed) {
-      return immediateFuture(fromCache);
-    }
-
     return transformAsync(client.validateDevice(Paths.device(device.getID())), r ->
       submitIfNotDisposed(shell, () -> validationCache.add(device, new DeviceValidationResult(r))));
   }
 
-  public DeviceValidationResult getValidationStatus(Device.Instance device) {
+  public DeviceValidationResult getCachedValidationStatus(Device.Instance device) {
```
[the remark this answers is not in the document]
Comment (ttanatb) [reply gapic/src/main/com/google/gapid/models/Devices.java:201]: That suggests we haven't seen the device before, so:
- When replaying a trace, it'll show you the device table dialog
- When capturing a new trace, you can pick a device from the drop down and validate accordingly

[the remark this answers is not in the document]
Comment (yalcinmelihyasin) [reply gapic/src/main/com/google/gapid/models/Devices.java:201]: :+1:

**Review by yalcinmelihyasin — APPROVED** [review]

File: gapic/src/main/com/google/gapid/views/DeviceValidationView.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+import org.eclipse.swt.program.Program;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.swt.widgets.Link;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.swt.SWT;
+
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Logger;
+
+public class DeviceValidationView extends Composite {
```
Comment (pmuetschard) [inline gapic/src/main/com/google/gapid/views/DeviceValidationView.java:48]: Consistent 2 space indentation please.

**Review by pmuetschard — APPROVED** [review]

**Pull request merged**